backend/migrations/3_polling_station_data_entries.sql

```diff
@@ -1,16 +1,8 @@
 CREATE TABLE polling_station_data_entries
 (
-    polling_station_id INTEGER NOT NULL,
-    entry_number       INTEGER NOT NULL,
-    progress           INTEGER NOT NULL,
-    data               BLOB,
-    client_state       BLOB,
-    updated_at         INTEGER NOT NULL DEFAULT (unixepoch()),
-    finalised_at       INTEGER,
+    polling_station_id INTEGER  PRIMARY KEY NOT NULL,
+    state              BLOB     NOT NULL,
+    updated_at         DATETIME NOT NULL DEFAULT CURRENT_TIMESTAMP,
```

Comment:
Why not keep the unix time? We had a discussion about this a while back.

SQLite doesn't have a `DATETIME` field, so this will be stored as a string now.

Reply:
I think we did this because it is only used in the frontend for display, and parsing/formatting the string this way with JS's `Date.toISOString()` is easier. See the line at the mention (don't know of a better way to point to a diff :) )

Reply:
I'd say `new Date(unix_epoch / 1000)` isn't that complicated either (and if we use unix time with millisecond resolution, we don't need the `/ 1000`).

Also, the `polling_station_results` table is still using unix time for the `created_at` column. I'd prefer to just stick to unix time.

Reply:
I'd like to quote RFC3339 for this:

> Date and time formats cause a lot of confusion and interoperability
> problems on the Internet.

Just to add my reasoning for doing this:
- Unix timestamps are much harder to read as a human (horrible to debug) and aren't always unambiguous, especially when a system is misconfigured to use local time for their unix timestamps somewhere. RFC3339 timestamps could of course still be generated wrongly when the system clock is misconfigured, but at least it will be obvious much quicker when looking at the data
- RFC3339 datetimes are unambiguous (although most datetime libraries (including the chrono and time crates in Rust) don't support leap seconds, but let's not worry about that for now, unix timestamps ignore them as well)
- Our Rust code shouldn't use all those integers, but properly typed data (which is what this PR is all about)
- Sqlite doesn't do much with datatypes in table definitions anyway, an integer column can store a string just as well (try it!), this is mostly type hints for us as humans. SQLite does specify the datetime column type, it just stores it as 'whatever', but it does that for all columns anyway. The only thing you get with an integer column is that it tries some conversion shenanigans for you (and if those conversions fail it will still fall back to whichever of the five storage types of sqlite can store the data).
- The size difference between the integer and the string is pretty small anyway, a fully specified RFC3339 UTC timestamp without milliseconds is only 20 bytes of string, only 2,5 times as big as a 64 bit unix timestamp
- `polling_station_results` should most definitely also not use unix timestamps if it is up to me

We should just keep the unix timestamps to kernel interfaces and system-internal timestamps, and for the rest of us: please always use RFC3339 when sending timestamps over the network or when storing timestamps in a portable way! This was my unix timestamps are bad public service announcement, now I'm back on holiday :wave: :beach_umbrella:

Reply:
Thanks for your convincing PSA 😃 Let's switch to using RFC3339. This PR is good to go for that part and I've created #782 to switch over `polling_station_results` too.

Enjoy the rest of your holiday!